src/ostree/ot-builtin-commit.c

```diff
@@ -582,6 +651,10 @@ ostree_builtin_commit (int argc, char **argv, GCancellable *cancellable, GError
     {
       gboolean update_summary;
       guint64 timestamp;
+      g_autoptr(GVariant) old_metadata = g_steal_pointer (&metadata);
+
+      fill_bindings (repo, old_metadata, &metadata);
```

Comment:
That's OK, let's just be sure we keep it in mind, since there are a lot of things that use the API directly and not the cmdline.